spec/models/rook_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+RSpec.describe Rook, type: :model do
+  describe "#valid_move?" do
+    let(:rook) { Rook.create(row: 0, col: 1) }
+
+    it "should allow legal vertical moves" do
```

Comment:
When writing the actual program, it's good to make it as simple as possible and still work. With tests, you have to balance simplicity with making sure you're testing enough to make sure the program works. But again, how to balance it is a matter of opinion.